tests/libs/test_matrix.py

```diff
@@ -134,7 +141,25 @@ def test_deserialize_messages_that_is_too_big(request_monitoring_message, capsys
     data = str(b"/0" * 1000000)
     assert sys.getsizeof(data) >= 1000000
 
-    deserialize_messages(data=data, peer_address=request_monitoring_message.sender)
+    deserialize_messages(
+        data=data,
+        peer_address=request_monitoring_message.sender,
+        rate_limiter=RateLimiter(allowed_bytes=1000000, reset_interval=timedelta(minutes=1)),
+    )
     captured = capsys.readouterr()
 
-    assert "Received message that is too big" in captured.out
+    assert "Sender is rate limited" in captured.out
+
+
+def test_rate_limiter():
+    limiter = RateLimiter(allowed_bytes=100, reset_interval=timedelta(seconds=1))
+    sender = Address(b"1" * 20)
+    for _ in range(50):
+        assert limiter.check_and_count(sender, 2)
```

Comment:
adding kwargs here would make it easier to understand that 2 is the number of bytes, not the value IMO